sklearn/utils/_array_api.py

```diff
@@ -357,6 +368,35 @@ def _expit(X):
     return 1.0 / (1.0 + xp.exp(-X))
 
 
+def _isneginf(X):
```

Comment:
TODO for Tim: open issue in Array API about asymmetry between `isinf` and `isneginf`